include/cantera/numerics/Func1.h

```diff
@@ -265,6 +274,68 @@ class Pow1 : public Func1
     virtual Func1& derivative() const;
 };
 
+
+//! implements a tabulated function
+class Tabulated1 : public Func1
+{
+public:
+    Tabulated1(const std::vector<double>& tvec, const std::vector<double>& fvec,
+               const std::string& method = "linear") :
+        Func1() {
+        if (tvec.size() != fvec.size()) {
+            throw CanteraError("Tabulated1::Tabulated1",
+                               "sizes of vectors do not match ({} vs {}).",
+                               tvec.size(), fvec.size());
+        } else if (tvec.empty()) {
+            throw CanteraError("Tabulated1::Tabulated1",
+                               "vectors must not be empty.");
+        }
+        m_tvec = tvec;
+        m_fvec = fvec;
```

Comment:
Presumably this should do some check to make sure that the time values are monotonically increasing, right?